### DISCUSSION TITLE
Anyone had reliable success with an LG 34WK95U-W + M1 Max MacBook Pro via Thunderbolt 3?

### DISCUSSION THREAD

**Discussion by VirtualWolf** [MonitorControl/MonitorControl#988] [Q&A] [answered]
I've got a 16" M1 Max MacBook Pro connected to my LG 34WK95U-W monitor via Thunderbolt 3, and I've found it seems extremely intermittent as to whether MonitorControl is able to control the display or not... the issue I have is that the DDC control doesn't work, but not _all_ the time, just mostly. It used to work, then stopped, and I'd ended up giving up fighting with it last time and moved the "Combined dimming switchover point" slider one notch short of the very right-hand side so it'd mostly just do software dimming.

After upgrading to macOS 12.13.1 yesterday, I found that when I hit the brightness down button the visual brightness was dropping all the way to 50%, despite going down only one notch on the OSD. I flipped the switchover point slider to the middle, and suddenly everything was working as normal again, huzzah!

However, today it's back to misbehaving again. :( I compiled [m1ddc](https://github.com/waydabber/m1ddc) to have a bit of a poke around and I'm kind of confused at the output. It clearly picks up the display:

```
[virtualwolf@shade:~/Downloads/m1ddc]$ ./m1ddc display list
1 - LG HDR 5K (1E6D2177-0000-0000-091F-0104B5502178)
```

But then apparently says it doesn't exist?

```
[virtualwolf@shade:~/Downloads/m1ddc]$ ./m1ddc get luminance
Could not find a suitable external display.
[virtualwolf@shade:~/Downloads/m1ddc]$ ./m1ddc display 1 get luminance
I2C communication failure: (iokit/common) no such device
```

This happens regardless of whether it's using DisplayPort 1.2 or 1.4. Has anyone else had any luck with this setup?

I just tested with my work-supplied 16" 2019 Intel MacBook Pro, and MonitorControl *does* work perfectly on that and controls the actual screen brightness via DDC (checked by making changes via MonitorControl and then seeing if the brightness the monitor reports has changed), and [ddcctl](https://github.com/kfix/ddcctl) can control the brightness from the CLI.

**Comment by VirtualWolf** [comment, marked as answer]
So following @patrickcwtsui's comment [here](https://github.com/MonitorControl/MonitorControl/issues/972#issuecomment-1064682500), I tried nuking `/Library/Preferences/com.apple.windowserver.displays.plist` and `~/Library/Preferences/ByHost/com.apple.windowserver.*` and restarting, and now MonitorControl is working again and actually controlling the monitor properly via DDC!

I'll be interested to see if this remains working or if it craps out again in future... 🤔

**Reply by patrickcwtsui** [reply]
In my experience it’s working well but there are times where I have to unplug the monitor to reset it, but that’s happened once in the last month (or even longer) - BUT my setup also includes an Apple BT keyboard and mouse so that might also affect the connection…

**Reply by VirtualWolf** [reply]
It indeed crapped out again, unplugging the monitor entirely from the power didn't help, I ended up having to restart the MBP itself but at least it's working again now without additional faffing around.
